### PR TITLE
#26 EA-1959, EA-1960 implemented KeyCloack authentication.

### PR DESCRIPTION
**Pull request by rgraf**
Updated create, retrieve, update and delete methods.

### CODE REVIEW

File: set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java [hunk 1]
```diff
@@ -1,124 +1,102 @@
...
+	public AuthenticationService getAuthenticationService() {
+		return authenticationService;
+	}
+
+	public void setAuthenticationService(AuthenticationService authenticationService) {
+		this.authenticationService = authenticationService;
+	}
+	
+	//verify user privileges
+	protected boolean hasPermission(Agent user, String operationName) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java:50]: is this method still used, if not, please remove it

Comment (gsergiu) [reply set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java:50]: Please also remove isAdmin adn isTester methods if they are not used anymore. Check also the Agent class if it is used

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java [hunk 2]
```diff
@@ -202,72 +140,28 @@ protected String serializeUserSet(LdProfiles profile, UserSet storedUserSet) thr
...
-			throws ApplicationAuthenticationException {
-		int modified = userSet.getModified().hashCode();
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(int etag, HttpServletRequest request) throws HttpException {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:150]: this method is now implemented in API Commons, please remove it

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java [hunk 3]
```diff
@@ -202,72 +140,28 @@ protected String serializeUserSet(LdProfiles profile, UserSet storedUserSet) thr
...
-	 * 
-	 * @param userSet
-	 * @param wsKey
-	 * @param queryUser
-	 * @throws ApplicationAuthenticationException
-	 */
-	public void hasModifyRights(UserSet userSet, String wsKey, String queryUser)
-			throws OperationAuthorizationException {
-
-		if (!(isAdmin(wsKey, queryUser) || userSet.getCreator().getName().equals(queryUser))) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:255]: the admin or owner verification is still required here, see the equivalent implementation in the annotation api

File: set-web/src/main/java/eu/europeana/set/web/service/controller/exception/GlobalExceptionHandler.java [hunk 4]
```diff
@@ -1,37 +1,44 @@
...
+	}
+	
+	@Override
+	protected ApiResponse getErrorReport(String apiKey, String action, Throwable th, boolean includeErrorStack) {
+		UserSetOperationResponse response = (UserSetOperationResponse) super.getErrorReport(apiKey, action, th, includeErrorStack);
+		return response;
+	}
+
+	@Override
+	protected AuthorizationService getAuthorizationService() {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/exception/GlobalExceptionHandler.java:39]: this method is not needed, please remove it

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 5]
```diff
@@ -56,40 +56,33 @@
...
 	public ResponseEntity<String> createUserSet(
-			@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
 			@RequestBody String userSet,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,			
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request)
 					throws HttpException {
 				
-		return storeUserSet(wskey, userSet, userToken, profile, request);
+		return storeUserSet(userSet, profile, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:64]: the verifyWriteAccess method needs to be invoked here

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 6]
```diff
@@ -56,40 +56,33 @@
...
-	protected ResponseEntity<String> storeUserSet(String wsKey, String userSetJsonLdStr, String userToken,
+	protected ResponseEntity<String> storeUserSet(String userSetJsonLdStr, 
 			String profileStr, HttpServletRequest request) throws HttpException {
 		try {
 			// validate user - check user credentials (all registered users can create) 
 			// if invalid respond with HTTP 401 or if unauthorized respond with HTTP 403;
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			Authentication authentication = verifyWriteAccess(Operations.CREATE, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:81]: this call needs to be moved the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 7]
```diff
@@ -152,34 +145,31 @@
...
 	public ResponseEntity<String> getUserSet(
-			@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String wskey,
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,			
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request) throws HttpException {
 
 		String action = "get:/set/{identifier}{.jsonld}";
-		return getUserSet(wskey, profile, identifier, request, action);
+		return getUserSet(profile, identifier, request, action);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:153]: the verify read access method must be invoked here, the wskey param is still available in the read methods

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 8]
```diff
@@ -152,34 +145,31 @@
...
-	private ResponseEntity<String> getUserSet(String wsKey, String profileStr, String identifier, 
+	private ResponseEntity<String> getUserSet(String profileStr, String identifier, 
 			HttpServletRequest request, String action)
 					throws HttpException {
 		try {
 			// check user credentials, if invalid respond with HTTP 401.
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+		    verifyReadAccess(request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:172]: this method must be invoked in the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 9]
```diff
@@ -217,55 +207,44 @@
...
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
 			@RequestBody String userSet,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "put:/set/{identifier}";
-		return updateUserSet(request, wskey, identifier, userSet, userToken, profile, action);
+		return updateUserSet(request, identifier, userSet, profile, action);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:218]: the verifyWriteAccess method needs to be invoked here

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 10]
```diff
@@ -217,55 +207,44 @@
...
+	protected ResponseEntity<String> updateUserSet(HttpServletRequest request, String identifier,
+			String userSetJsonLdStr, String profileStr, String action) throws HttpException {
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
-			// check client access (a valid "wskey" must be provided)
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			verifyWriteAccess(Operations.UPDATE, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:238]: this method needs to be invoked in the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 11]
```diff
@@ -351,18 +330,17 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_LOCAL_ID) String localId,
 			@RequestParam(value = WebUserSetFields.PATH_PARAM_POSITION, required=false) String position,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "put:/set/{identifier}/{dataset_id}/{local_id}?position=POSITION";
-		return insertItemIntoUserSet(request, wskey, identifier, datasetId, localId, position, userToken, 
+		return insertItemIntoUserSet(request, identifier, datasetId, localId, position,
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:343]: the verifyWriteAccess method needs to be invoked here

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 12]
```diff
@@ -371,42 +349,33 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
+			String identifier, String datasetId, String localId, String position,  
 			String profileStr, String action) throws HttpException {
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+			verifyWriteAccess(Operations.UPDATE, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:369]: this method needs to be invoked in the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 13]
```diff
@@ -447,24 +416,21 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_DATASET_ID) String datasetId,
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_LOCAL_ID) String localId,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
-//			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "get:/set/{identifier}/{dataset_id}/{local_id}";
-		return isItemInUserSet(request, wskey, identifier, datasetId, localId, userToken, action);
+		return isItemInUserSet(request, identifier, datasetId, localId, action);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:427]: please invoke the verifyReadAccess here, the wskey param must still be available in the GET methods

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 14]
```diff
@@ -473,18 +439,15 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
-
-			// authorize user
-			getAuthorizationService().authorizeUser(userToken, wsKey, identifier, Operations.RETRIEVE);
+		    // verify access rights
+			verifyReadAccess(request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:450]: this method needs to be invoked in the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 15]
```diff
@@ -533,24 +496,22 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_DATASET_ID) String datasetId,
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_LOCAL_ID) String localId,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			@RequestParam(value = WebUserSetFields.PROFILE, required = false, defaultValue = WebUserSetFields.PROFILE_MINIMAL) String profile,			
 			HttpServletRequest request
 			) throws HttpException {
 		
 		String action = "delete:/set/{identifier}/{dataset_id}/{local_id}";
-		return deleteItemFromUserSet(request, wskey, identifier, datasetId, localId, userToken, profile, action);
+		return deleteItemFromUserSet(request, identifier, datasetId, localId, profile, action);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:508]: the verifyWriteAccess method needs to be invoked here

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 16]
```diff
@@ -560,29 +521,22 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
+			String identifier, String datasetId, String localId, 
 			String profileStr, String action) throws HttpException {
 		
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
+		    // verify access rights
+		    verifyWriteAccess(Operations.DELETE, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:532]: the invokation of the write access needs to be invoked in the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 17]
```diff
@@ -642,86 +596,45 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
-			@RequestParam(value = WebUserSetFields.PARAM_WSKEY, required = false) String apiKey,
 			@PathVariable(value = WebUserSetFields.PATH_PARAM_SET_ID) String identifier,
-			@RequestParam(value = WebUserSetFields.USER_TOKEN, required = false, defaultValue = WebUserSetFields.USER_ANONYMOUNS) String userToken,
 			HttpServletRequest request
 			) throws HttpException {
 
-		userToken = getUserToken(userToken, request);
-				
-		return deleteUserSet(request, identifier, apiKey, userToken);
+		return deleteUserSet(request, identifier);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:603]: the verifyWriteAccess method needs to be invoked here

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 18]
```diff
@@ -642,86 +596,45 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
-			// check client access (a valid "wskey" must be provided)
-			// Check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
-
-			// authorize user or if unauthorized respond with HTTP 403
-			getAuthorizationService().authorizeUser(userToken, wsKey, identifier, Operations.DELETE);
+		    // verify access rights
+		    verifyWriteAccess(Operations.DELETE, request);
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:618]: needs to be moved to the calling method

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java [hunk 19]
```diff
@@ -642,86 +596,45 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
...
-  					 if (isOwner(existingUserSet, userToken)) {
-  						 getUserSetService().disableUserSet(existingUserSet);
-  					 } else {
- 						// if the user is a registered user but not the owner, the response should be 401 (unathorized)
- 						httpStatus = HttpStatus.UNAUTHORIZED;					
-  					 }
-				 }
-			}			
+
+		    if (existingUserSet.isDisabled()) {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:630]: please confirm if this behaviour was added to specs

File: set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java [hunk 20]
```diff
@@ -0,0 +1,35 @@
+package eu.europeana.set.web.exception.request;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+
+import eu.europeana.api.common.config.I18nConstants;
+import eu.europeana.api.commons.web.exception.HttpException;
+
+
+public class HeaderValidationException extends HttpException {
```
[the remark this answers is not in the document]
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java:10]: In my verion of set-api this class is not presented

[the remark this answers is not in the document]
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java:10]: It is in api-commons

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java [hunk 21]
```diff
@@ -0,0 +1,74 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import eu.europeana.api.commons.definitions.vocabulary.Role;
+
+public enum UserRoles implements Role {
+
+	anonimous(new String[]{Operations.RETRIEVE}), 
+	user(new String[]{Operations.RETRIEVE, Operations.CREATE, Operations.DELETE, Operations.UPDATE, Operations.REPORT}),
```
[the remark this answers is not in the document]
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java:12]: There is no UserRoles in my version. But I removed REPORT from Options and Roles

File: set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java (hunk 1 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/authorization/AuthorizationServiceImpl.java:50]: removed mentioned methods. Agent ist used in almost 400 cases.

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java (hunk 2 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:150]: This method is already not in BaseRest

File: set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java (hunk 3 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/BaseRest.java:255]: took implementation from annotation api

File: set-web/src/main/java/eu/europeana/set/web/service/controller/exception/GlobalExceptionHandler.java (hunk 4 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/exception/GlobalExceptionHandler.java]: File removed

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 5 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 6 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: removed validateApiKey call

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 7 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyReadAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 8 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyReadAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 9 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 10 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 11 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 12 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 13 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyReadAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 14 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyReadAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 15 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 16 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 17 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 18 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java]: verifyWriteAccess invoked as suggested

File: set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java (hunk 19 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java:664]: behaviour remains the same

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java [hunk 22]
```diff
@@ -1,83 +1,85 @@
...
+
+	// Web application
+	public static final String USER_SET_PROVIDER = "userset";
+	public static final String USER_SET_APPLICATION_NAME = "webuserset";
+	public static final String SLASH = "/";
+	public static final String PAR_CHAR = "?";
+	public static final String AND = "&";
+	public static final String EQUALS = "=";
+	public static final String JSON_LD_REST = ".jsonld";
+	public static final String FORMAT_JSONLD = "jsonld";
```
Comment (gsergiu) [inline set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java:19]: Please Move this constant to a base class in API Commons (WebFields) and remode the duplications in Set, Annotations and Entity API

File: set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java (hunk 20 above)
[the remark this answers is not in the document]
Comment (gsergiu) [reply set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java:10]: I mean the HeaderValidationException  not the HttpException

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java [hunk 23]
```diff
@@ -1,27 +1,24 @@
...
+package eu.europeana.set.web.model.vocabulary;
+
+public interface Operations {
+
+	//search is also retrieve
+	//crud operations
+	public static final String RETRIEVE = "retrieve";
+	public static final String CREATE = "create";
+	public static final String UPDATE = "update";
+	public static final String DELETE = "delete";
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java:10]: must extend the Commons-API class and remove dupplicaed constants: https://github.com/europeana/api-commons/blob/ff79e7c6f0a0c71b406f184da1784dd391bc4a51/commons-web/src/main/java/eu/europeana/api/commons/web/model/vocabulary/Operations.java

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java [hunk 24]
```diff
@@ -0,0 +1,74 @@
+package eu.europeana.set.web.model.vocabulary;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import eu.europeana.api.commons.definitions.vocabulary.Role;
+
+public enum UserRoles implements Role {
```
Comment (gsergiu) [inline set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java:9]: Is this class not a dupplication of the upper Roles enum? I assume that the upper one needs to be deleted

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java (hunk 22 above)
Comment (rgraf) [reply set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java:19]: Moved for set-api

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java [hunk 25]
```diff
@@ -1,83 +1,85 @@
...
+	public static final String USER_ADMIN = "admin";	
+	public static final String PROVIDER_EUROPEANA_DEV = "eanadev";	
+	
+	//
+	// Validation definitions
+	//
+	public static final String READ_METHOD = "read";
+	public static final String WRITE_METHOD = "write";
+	public static final String DELETE_METHOD = "delete";	
+	public static final String DEFAULT_CREATOR_URL = "http://data.europeana.eu/user/";
```
[the remark this answers is not in the document]
Comment (rgraf) [reply set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java:84]: Moved for set-api

File: set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java (hunk 20 above)
[the remark this answers is not in the document]
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/exception/request/HeaderValidationException.java:10]: I know. It was done. Just this branch is totally corrupted. I moved everything to a new branch: https://github.com/europeana/set-api/tree/%2326_fix_keycloak_issues

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java (hunk 23 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java:10]: Extended with Operations from api-commons and removed duplicates in set-api

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java (hunk 24 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/model/vocabulary/UserRoles.java:9]: This class is deleted. Just this branch is totally corrupted. I moved everything to a new branch: https://github.com/europeana/set-api/tree/%2326_fix_keycloak_issues

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java (hunk 22 above)
Comment (rgraf) [reply set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java:19]: Moved for entity-api

File: set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java (hunk 25 above)
[the remark this answers is not in the document]
Comment (rgraf) [reply set-definitions/src/main/java/eu/europeana/set/definitions/model/vocabulary/WebUserSetFields.java:84]: Moved for entity-api

File: set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java (hunk 23 above)
Comment (rgraf) [reply set-web/src/main/java/eu/europeana/set/web/model/vocabulary/Operations.java:10]: The same done for entity-api

**Comment by gsergiu** [comment]
new pul request was created including these changes

**Pull request closed without merging**